### DISCUSSION TITLE
getServerSideProps or on demand ISR for high dynamic data

### DISCUSSION THREAD

**Discussion by bairrada97** [vercel/community#528] [Help]
Hello I would like to know what are the best approach here in order to have the highest perfomance on page load.

I have high dynamic data that needs to be updated every 15 seconds,  just a basic list with data.

Using next.js I tried using getStaticProps with `revalidate: 15 ` and setting up a revalidate function to revalidate every time there is a fetch call. so when on the client side there are a fetch call I also call a revalidate function, if user refreshes the page it will see the new updated content. Until here everything good and really fast perfomance. 

Problem comes that my data needs a timezone parameter this means that if a user with timezone 'Europe/Berlin' revalidate and other user from other side of the world enter the page will see the content of the user with timezone 'Europe/Berlin'.

So I come to mind that what if I use getServerSideProps with Vercel cache control?

since for me its hard to test what user with different timezone see when load the page, do you think my approach using getServerSide props is the correct approach? or you have a better idea for my use case?

Edit: title name

**Comment by leerob** [comment]
You might consider using on-demand Incremental Static Regeneration: https://nextjs.org/blog/next-12-1#on-demand-incremental-static-regeneration-beta

**Reply by bairrada97** [reply]
This is what Im actually using, when I mention that im using a revalidate function its on demand ISR.
correct me if im wrong, since I cant test different timezones and what users see on the page, but the issue is since on demand ISR its pre rendered on build time a user with timezone 'Europe/Berlin' that revalidate the page will see the correct data but once a user from a different timezone enter the website it will see first the content from the user with 'Europe/Berlin timezone and then after 15 seconds updates to the content from his current timezone
